hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -7280,7 +7280,7 @@ public Map<byte[], List<Path>> bulkLoadHFiles(Collection<Pair<byte[], String>> f
               reqTmp ? null : fs.getRegionDir().toString());
           }
           Pair<Path, Path> pair = null;
-          if (reqTmp) {
+          if (reqTmp || (!StoreFileInfo.isHFile(finalPath))) {
```

Comment:
nit: maybe remove the extra `()` ?

```suggestion
          if (reqTmp || !StoreFileInfo.isHFile(finalPath)) {
```

do you have any stacktrace and example of the non-HFile ? also , when you said `has been split in the bulkload process` , what would this split generate?